src/app/qgsclipboard.cpp

```diff
@@ -131,14 +133,15 @@ void QgsClipboard::generateClipboardText( QString &textContent, QString &htmlCon
 
         for ( int idx = 0; idx < attributes.count(); ++idx )
         {
-          QString value = attributes.at( idx ).toString();
+          QString value = QString::fromStdString( QgsJsonUtils::jsonFromVariant( attributes.at( idx ) ).dump() );
```

Comment:
The diference is, in v3.24 all is copied without `"`, e.g.
```
wkt_geom	date	datetime	integerlist	decimallist	stringlist2	stringlist	integerlist2	decimallist2	integer	string
Point (7.60173582322795571 51.9631281667119751)	2022-02-22	04:00:00.000							4	abc
```
While with that jsonFromVariant, all except the number is copied with `"`
```
wkt_geom	date	datetime	integerlist	decimallist	stringlist2	stringlist	integerlist2	decimallist2	integer	string
Point (7.60173582322795571 51.9631281667119751)	"2022-02-22"	"04:00:00.000"	[2,1]	[0.9,0.0]	["c","b","a"]	["a","b","c"]	[2,1]	[0.9,0.0]	4	"abc"
```

The second one seems more consistent. We can differentiate a number and non-number here.

But after reading your comment, perhaps the original purpose is to copy the value without the `"`. Is that correct? If so, I will update it. But not sure how it works with JSON data type. Should it include the full format like:
```
{"date":"3333","def":"ghi","test":"abc"}
```
or a simple one:
```
date: 3333, def: ghi, test: abc
``` 

Thanks for the feedback!

Reply:
Hi @nyalldawson, after some thought, I decided to only use JSON-way if it's not convertible to String. Below are some examples:
```
wkt_geom	fid	json	val
Point (-0.7402777777777777 0.625)	1	{"date":"3333","def":"ghi","test":"abc"}	11111

wkt_geom	date	datetime	integerlist	decimallist	stringlist2	stringlist	integerlist2	decimallist2	integer	string
Point (7.60227984718220906 51.96190223848584822)	2022-03-02	00:00:00.000	[3,2,1]	[8.0,0.0]	["c","b","a"]	["z","y","x"]	null	[8.0,0.0]	1	ghi
```

I think it makes sense since we only want to copy the value, but for JSON and list, we still get the `"` char to differentiate between string or non-string (and make the JSON format valid)